### ISSUE TITLE
Key navigation for year and quarter picker

### ISSUE THREAD

**Issue by ThereseSande** [Hacker0x01/react-datepicker#3464]
**Describe the bug**
The year and quarter picker is missing some functionality for keyboard navigation. For the year picker you have to press the tab button twice to enter into the field for the years. Also, the years are listed in two columns, and it is easy to think that you can navigate up and down, but you can only navigate left and right with the arrow keys. When you navigate out of the year picker with the tab key, it does not close.
For the quarter picker it does not enter at all into the field where you can pick a quarter, you get "stuck" in the year bar of the picker.
**To Reproduce**
You can see the same behaviour in the https://reactdatepicker.com/

**Expected behavior**
The actual datepicker has the expected behaviour

**Desktop (please complete the following information):**

- chrome and firefox
